### PR TITLE
upgrade MERRA2 climatology from 2003-2014 mean to 2014 to 2024 mean

### PR DESCRIPTION
**Pull request by AnningCheng-NOAA**
## Commit Queue Requirements:

- [X] Fill out all sections of this template.
- [ ] All sub component pull requests have been reviewed by their code managers.
- [X] Run the full Intel+GNU RT suite (compared to current baselines) on either Hera/Derecho/Hercules
- [ ] Commit 'test_changes.list' from previous step
---
## Description:

upgrade MERRA2 climatology from 2003-2014 mean to 2014 to 2024 mean

### Commit Message:

```
* UFSWM - Upgrade MERRA2 climatology from 2003-2014 mean to 2014 to 2024 mean
```

### Priority:

* Normal

### UFSWM:

* None

### Sub component Pull Requests:

* None

### UFSWM Blocking Dependencies:

* None

---
### Regression Test Changes (Please commit test_changes.list):

* PR Updates/Changes Baselines.
All tests using MERRA2 climatology needs new baseline

### Input data Changes:

* New input data.
### Library Changes/Upgrades:

* No Updates
  
---

## Testing Log:
- RDHPCS
  - [x] Hera
  - 
[RegressionTests_hera.log](https://github.com/ufs-community/ufs-weather-model/files/15265977/RegressionTests_hera.log)

  - [x] Orion
  - [x] Hercules
  - [x] Jet
  - [x] Gaea
  - [x] Derecho
- WCOSS2
  - [ ] Dogwood/Cactus
  - [ ] Acorn
- [x] CI
- [ ] opnReqTest (complete task if unnecessary)

### CODE REVIEW

**Comment by zach1221** [comment]
@AnningCheng-NOAA can you please sync up this PR branch for us? We're looking to begin testing this PR next.

Also, please add a commit message to the template.

**Comment by zach1221** [comment]
@BrianCurtis-NOAA once the PR is synced up, I'm going to re-run on Hera to confirm which cases are changing, as there are quite a few failures in the previously attached Hera log vs the tests_changes.list.

File: tests/fv3_conf/control_run.IN [hunk 1]
```diff
@@ -133,7 +133,7 @@ fi
 #merra2
 if [ $USE_MERRA2 = .true. ]; then
   for n in 01 02 03 04 05 06 07 08 09 10 11 12; do
-    cp @[INPUTDATA_ROOT]/FV3_input_data_INCCN_aeroclim/MERRA2/merra2.aerclim.2003-2014.m${n}.nc aeroclim.m${n}.nc
+    cp /scratch1/NCEPDEV/global/Anning.Cheng/MERRA2_y14_24/merra2_1423_${n}.nc aeroclim.m${n}.nc
```
Comment (DusanJovic-NOAA) [inline tests/fv3_conf/control_run.IN]: The input data should be copied from INPUTDATA_ROOT, not personal directories.

**Comment by AnningCheng-NOAA** [comment]
I do not have the right to access  @[INPUTDATA_ROOT.  Could anybody ro
resolve this issue?

**Comment by zach1221** [comment]
[RegressionTests_hera.log](https://github.com/ufs-community/ufs-weather-model/files/15407418/RegressionTests_hera.log)

**Comment by BrianCurtis-NOAA** [comment]
@zach1221 You're missing two things, one was the label which I took care of and two, addressing any remaining issues in the PR. For the second item, you'll need to let @AnningCheng-NOAA know where the new input data was placed, so they can update their repository with the updated location. Thus this PR is not ready for the commit queue quite yet.

**Comment by BrianCurtis-NOAA** [comment]
@zach1221 Also we'll need an updated test_changes.list

**Comment by zach1221** [comment]
> @zach1221 Also we'll need an updated test_changes.list

Ok, I can add a test change log, I wasn't aware this was my responsibility. I will do this going forward if it's not included.

**Comment by BrianCurtis-NOAA** [comment]
> > @zach1221 Also we'll need an updated test_changes.list
> 
> Ok, I can add a test change log, I wasn't aware this was my responsibility. I will do this going forward if it's not included.

This was only because you ran the Hera pre-test and there was no test_changes.list changes pushed to this branch. Normally that would be on the PR authors plate.

**Comment by BrianCurtis-NOAA** [comment]
@zach1221 are you using `-b test_changes.list` in your work for this PR ?

**Comment by FernandoAndrade-NOAA** [comment]
> @zach1221 are you using `-b test_changes.list` in your work for this PR ?

I think he created a temp rt.conf with just the changed tests. Running with `-b` resulted in rt.sh thinking the tests were not available to run on jet and gaea from my side unless I ran with the full  `-l rt.conf` instead.

**Comment by BrianCurtis-NOAA** [comment]
> > @zach1221 are you using `-b test_changes.list` in your work for this PR ?
> 
> I think he created a temp rt.conf with just the changed tests. Running with `-b` resulted in rt.sh thinking the tests were not available to run on jet and gaea from my side unless I ran with the full `-l rt.conf` instead.

I use it each PR, and this PR caused the issue you describe for me. Would I be correct in thinking the test_changes.list was edited in an editor and not coming from the generated one from rt.sh? I can't seem to find out why this PR is causing the issues, and that is my guess at this point.

**Comment by FernandoAndrade-NOAA** [comment]
> > > @zach1221 are you using `-b test_changes.list` in your work for this PR ?
> > 
> > 
> > I think he created a temp rt.conf with just the changed tests. Running with `-b` resulted in rt.sh thinking the tests were not available to run on jet and gaea from my side unless I ran with the full `-l rt.conf` instead.
> 
> I use it each PR, and this PR caused the issue you describe for me. Would I be correct in thinking the test_changes.list was edited in an editor and not coming from the generated one from rt.sh? I can't seem to find out why this PR is causing the issues, and that is my guess at this point.

Would manually edited tests into test_changes.list cause an issue with -b's behavior? If so it may be worth seeing if future improvements could allow it to accept the listed tests in any order. It's a bit tricky to diagnose, I'm not getting any real error output other than rt.sh rejecting the specified tests and exiting.

**Comment by BrianCurtis-NOAA** [comment]
> > > > @zach1221 are you using `-b test_changes.list` in your work for this PR ?
> > > 
> > > 
> > > I think he created a temp rt.conf with just the changed tests. Running with `-b` resulted in rt.sh thinking the tests were not available to run on jet and gaea from my side unless I ran with the full `-l rt.conf` instead.
> > 
> > 
> > I use it each PR, and this PR caused the issue you describe for me. Would I be correct in thinking the test_changes.list was edited in an editor and not coming from the generated one from rt.sh? I can't seem to find out why this PR is causing the issues, and that is my guess at this point.
> 
> Would manually edited tests into test_changes.list cause an issue with -b's behavior? If so it may be worth seeing if future improvements could allow it to accept the listed tests in any order. It's a bit tricky to diagnose, I'm not getting any real error output other than rt.sh rejecting the specified tests and exiting.

Right, me either. But rt.sh hates that file, somehow.

**Comment by BrianCurtis-NOAA** [comment]
I've compared two test_changes.list and there is a difference:
```
brian.curtis@alogin01:/lfs/h1/emc/nems/noscrub/brian.curtis/git/AnningCheng-NOAA/ufs-weather-model/tests> cat -A ../../../BrianCurtis-NOAA/ufs-weather-model/develop/tests/test_changes.list 
cpld_control_p8_mixedmode intel$
cpld_control_gfsv17 intel$
cpld_control_gfsv17_iau intel$
cpld_restart_gfsv17 intel$
cpld_mpi_gfsv17 intel$
cpld_debug_gfsv17 intel$
cpld_control_p8 intel$
cpld_control_p8.v2.sfc intel$
cpld_restart_p8 intel$
cpld_control_qr_p8 intel$
cpld_restart_qr_p8 intel$
cpld_2threads_p8 intel$
cpld_decomp_p8 intel$
cpld_mpi_p8 intel$
cpld_control_ciceC_p8 intel$
cpld_control_c192_p8 intel$
cpld_restart_c192_p8 intel$
cpld_bmark_p8 intel$
cpld_restart_bmark_p8 intel$
cpld_s2sa_p8 intel$
cpld_control_noaero_p8 intel$
cpld_control_nowave_noaero_p8 intel$
cpld_debug_p8 intel$
cpld_debug_noaero_p8 intel$
cpld_control_noaero_p8_agrid intel$
cpld_control_c48 intel$
cpld_control_p8_faster intel$
cpld_control_pdlib_p8 intel$
cpld_restart_pdlib_p8 intel$
cpld_mpi_pdlib_p8 intel$
cpld_debug_pdlib_p8 intel$
control_flake intel$
control_CubedSphereGrid intel$
control_CubedSphereGrid_parallel intel$
control_latlon intel$
control_wrtGauss_netcdf_parallel intel$
control_c48 intel$
control_c48.v2.sfc intel$
control_c192 intel$
control_c384 intel$
control_c384gdas intel$
control_stochy intel$
control_stochy_restart intel$
control_lndp intel$
control_iovr4 intel$
control_iovr5 intel$
control_p8 intel$
control_p8.v2.sfc intel$
control_p8_ugwpv1 intel$
control_restart_p8 intel$
control_noqr_p8 intel$
control_restart_noqr_p8 intel$
control_decomp_p8 intel$
control_2threads_p8 intel$
control_p8_lndp intel$
control_p8_rrtmgp intel$
control_p8_mynn intel$
merra2_thompson intel$
rap_control intel$
rap_decomp intel$
rap_2threads intel$
rap_restart intel$
rap_sfcdiff intel$
rap_sfcdiff_decomp intel$
rap_sfcdiff_restart intel$
hrrr_control intel$
hrrr_control_decomp intel$
hrrr_control_2threads intel$
hrrr_control_restart intel$
rrfs_v1beta intel$
rrfs_v1nssl intel$
rrfs_v1nssl_nohailnoccn intel$
control_csawmg intel$
control_csawmgt intel$
control_ras intel$
control_wam intel$
control_p8_faster intel$
control_CubedSphereGrid_debug intel$
control_wrtGauss_netcdf_parallel_debug intel$
control_stochy_debug intel$
control_lndp_debug intel$
control_csawmg_debug intel$
control_csawmgt_debug intel$
control_ras_debug intel$
control_diag_debug intel$
control_debug_p8 intel$
rap_control_debug intel$
hrrr_control_debug intel$
hrrr_gf_debug intel$
hrrr_c3_debug intel$
rap_unified_drag_suite_debug intel$
rap_diag_debug intel$
rap_cires_ugwp_debug intel$
rap_unified_ugwp_debug intel$
rap_lndp_debug intel$
rap_progcld_thompson_debug intel$
rap_noah_debug intel$
rap_sfcdiff_debug intel$
rap_noah_sfcdiff_cires_ugwp_debug intel$
rrfs_v1beta_debug intel$
rap_clm_lake_debug intel$
rap_flake_debug intel$
gnv1_c96_no_nest_debug intel$
control_wam_debug intel$
rap_control_dyn32_phy32 intel$
hrrr_control_dyn32_phy32 intel$
rap_2threads_dyn32_phy32 intel$
hrrr_control_2threads_dyn32_phy32 intel$
hrrr_control_decomp_dyn32_phy32 intel$
rap_restart_dyn32_phy32 intel$
hrrr_control_restart_dyn32_phy32 intel$
conus13km_control intel$
conus13km_2threads intel$
conus13km_restart_mismatch intel$
rap_control_dyn64_phy32 intel$
rap_control_debug_dyn32_phy32 intel$
hrrr_control_debug_dyn32_phy32 intel$
conus13km_debug intel$
conus13km_debug_qr intel$
conus13km_debug_2threads intel$
conus13km_radar_tten_debug intel$
rap_control_dyn64_phy32_debug intel$
hafs_global_1nest_atm intel$
hafs_global_multiple_4nests_atm intel$
hafs_global_storm_following_1nest_atm intel$
gnv1_nested intel$
control_p8_atmlnd_sbs intel$
control_p8_atmlnd intel$
control_restart_p8_atmlnd intel$
atmwav_control_noaero_p8 intel$
control_atmwav intel$
atmaero_control_p8 intel$
atmaero_control_p8_rad intel$
atmaero_control_p8_rad_micro intel$
control_c48 gnu$
control_stochy gnu$
control_ras gnu$
control_p8 gnu$
control_p8_ugwpv1 gnu$
control_flake gnu$
rap_control gnu$
rap_decomp gnu$
rap_2threads gnu$
rap_restart gnu$
rap_sfcdiff gnu$
rap_sfcdiff_decomp gnu$
rap_sfcdiff_restart gnu$
hrrr_control gnu$
hrrr_control_noqr gnu$
hrrr_control_2threads gnu$
hrrr_control_decomp gnu$
hrrr_control_restart gnu$
hrrr_control_restart_noqr gnu$
rrfs_v1beta gnu$
control_diag_debug gnu$
rap_control_debug gnu$
hrrr_control_debug gnu$
hrrr_gf_debug gnu$
hrrr_c3_debug gnu$
rap_diag_debug gnu$
rap_noah_sfcdiff_cires_ugwp_debug gnu$
rap_progcld_thompson_debug gnu$
rrfs_v1beta_debug gnu$
control_ras_debug gnu$
control_stochy_debug gnu$
control_debug_p8 gnu$
rap_flake_debug gnu$
rap_clm_lake_debug gnu$
gnv1_c96_no_nest_debug gnu$
control_wam_debug gnu$
rap_control_dyn32_phy32 gnu$
hrrr_control_dyn32_phy32 gnu$
rap_2threads_dyn32_phy32 gnu$
hrrr_control_2threads_dyn32_phy32 gnu$
hrrr_control_decomp_dyn32_phy32 gnu$
rap_restart_dyn32_phy32 gnu$
hrrr_control_restart_dyn32_phy32 gnu$
conus13km_control gnu$
conus13km_2threads gnu$
conus13km_restart_mismatch gnu$
rap_control_dyn64_phy32 gnu$
rap_control_debug_dyn32_phy32 gnu$
hrrr_control_debug_dyn32_phy32 gnu$
conus13km_debug gnu$
conus13km_debug_qr gnu$
conus13km_debug_2threads gnu$
conus13km_radar_tten_debug gnu$
rap_control_dyn64_phy32_debug gnu$
cpld_control_p8 gnu$
cpld_control_nowave_noaero_p8 gnu$
cpld_debug_p8 gnu$
cpld_control_pdlib_p8 gnu$
cpld_debug_pdlib_p8 gnu$
brian.curtis@alogin01:/lfs/h1/emc/nems/noscrub/brian.curtis/git/AnningCheng-NOAA/ufs-weather-model/tests> cat -A test_changes.list
cpld_control_p8_mixedmode intel^M$
cpld_control_gfsv17 intel^M$
^M$
```
I need to know what causes the ^M that the `cat -A` sees.

**Comment by BrianCurtis-NOAA** [comment]
OK update: ^M is from a Windows editor. I've cleaned that out on my end and it works OK.

@jkbk2004 @zach1221 @FernandoAndrade-NOAA Please take note of this.

**Comment by FernandoAndrade-NOAA** [comment]
> OK update: ^M is from a Windows editor. I've cleaned that out on my end and it works OK.
> 
> @jkbk2004 @zach1221 @FernandoAndrade-NOAA Please take note of this.

Would this be caused by not specifying LF in a windows editor?

**Comment by FernandoAndrade-NOAA** [comment]
> > OK update: ^M is from a Windows editor. I've cleaned that out on my end and it works OK.
> > @jkbk2004 @zach1221 @FernandoAndrade-NOAA Please take note of this.
> 
> Would this be caused by not specifying LF in a windows editor?

Never mind, your commit message pretty much confirms that, nice catch!

**Comment by BrianCurtis-NOAA** [comment]
Hey @SamuelTrahanNOAA I'm getting failures on Acorn for IFI tests (control decomp 2threads). I'm only curious if these changes would cause baseline changes in the IFI tests. If yes, no problem and i'll generate new baselines. If no, then we need to figure out why all files in the test are not identical in comparison.

**Comment by FernandoAndrade-NOAA** [comment]
@AnningCheng-NOAA we are getting close to starting the merge process, please be sure to fill out the Commit Message section of the PR template.

**Comment by jkbk2004** [comment]
> Hey @SamuelTrahanNOAA I'm getting failures on Acorn for IFI tests (control decomp 2threads). I'm only curious if these changes would cause baseline changes in the IFI tests. If yes, no problem and i'll generate new baselines. If no, then we need to figure out why all files in the test are not identical in comparison.

https://github.com/ufs-community/ufs-weather-model/blob/develop/tests/tests/regional_ifi_control#L22

**Comment by BrianCurtis-NOAA** [comment]
> > Hey @SamuelTrahanNOAA I'm getting failures on Acorn for IFI tests (control decomp 2threads). I'm only curious if these changes would cause baseline changes in the IFI tests. If yes, no problem and i'll generate new baselines. If no, then we need to figure out why all files in the test are not identical in comparison.
> 
> https://github.com/ufs-community/ufs-weather-model/blob/develop/tests/tests/regional_ifi_control#L22

OK, somehow I thought there were physics changes to this. It makes full sense that the IFI tests are changing here. Thanks.

**Review by jkbk2004 — APPROVED** [review]

**Review by FernandoAndrade-NOAA — APPROVED** [review]

**Pull request merged**